front-end/src/pages/ProposalEdit/ProposalEdit.js

```diff
@@ -0,0 +1,648 @@
+                            <Grid item xs={4} container direction="column">
+                                <Grid item>
+                                    <CurrencyTextField
+                                        value={valueBudget}
+                                        currencySymbol="€"
+                                        outputFormat="string"
+                                        decimalCharacter="."
+                                        digitGroupSeparator=","
+                                        placeholder="Budget"
+                                        onChange={(event, value) => setValueBudget(value)}
+                                        InputProps={{
+                                            classes: {
+                                                input: classes.input
+                                            }
+                                        }}
+                                        className={classes.currencyInput}
+                                    />
+                                </Grid>
+                                <Grid item className="type-wrape">
+                                    <FormControl className={classes.formControl}>
+                                        <InputLabel htmlFor="type-select" className={classes.inputLabel}>Type</InputLabel>
+                                        <Select
+                                            native
+                                            value={state.type}
+                                            onChange={handleChange}
+                                            inputProps={{
+                                                name: 'type',
+                                                id: 'type-select',
+                                            }}
+                                            errors={errors}
+                                        >
+                                            <option aria-label="" value="" />
+                                            <option value="new">New</option>
+                                            <option value="upgrade">Upgrade</option>
+                                            <option value="remove">Remove</option>
+                                        </Select>
+                                        {hasErrorType && <FormHelperText>Please select a type.</FormHelperText>}
+                                    </FormControl>
+                                </Grid>
+                                <Grid item container className="category-wraper" direction="column" spacing={1}>
+                                    <Grid item>
+                                        <TitleCategoryTypography className={classes.categoryTitle}>Categories</TitleCategoryTypography>
+                                    </Grid>
+                                    <Grid item container>
+                                        <Grid item xs={6} container alignItems="center">
```

Comment:
Can we make this checkbox grid dynamic instead of manually writing 8 times ?
If we do this so we could avoid writing long react templates